refactor(transformer/arrow-function): `generate_super_binding_name` take `&str` and `&TraverseCtx`

overlookmotel opened the pull request:
Pure refactor. Take `&str` rather than `&Atom` (1 less indirection) and `&TraverseCtx<'a>` (more conventional).

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7310?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7312** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7312?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7311** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7311?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7310** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7310?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7310?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7234** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7234?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Nov 17, 12:01 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(transformer/arrow-function): `generate_super_binding_name` take `&str` and `&TraverseCtx` (#7310)

Pure refactor. Take `&str` rather than `&Atom` (1 less indirection) and `&TraverseCtx<'a>` (more conventional).

Dunqing commented:
I don't know why this conflicted

overlookmotel commented:
> I don't know why this conflicted

Graphite mangled my PR! It seems to get confused sometimes when downstack PR has multiple commits.
